install_sct

```diff
@@ -325,17 +325,6 @@ if [ "$(uname)" = "Darwin" ]; then
   (command -v realpath >/dev/null) || realpath() {
     python3 -c 'import sys, os; [print(os.path.realpath(f)) for f in sys.argv[1:]]' "$@"
   }
-
```

Comment:
Bizarrely, `test_batch_processing.sh` is now failing:

```python
___________ test_batch_processing_results[mt/mtr_in_wm.csv-0-MAP()] ____________
Traceback (most recent call last):
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/testing/batch_processing/test_batch_processing.py", line 42, in test_batch_processing_results
    assert (get_csv_float_value(csv_output, row, column) == pytest.approx(  # Default rel_tolerance: 1e-6
AssertionError: assert 54.42425821653468 == 54.37973987894469 ± 5.4e-05
  +54.42425821653468
  -54.37973987894469 ± 5.4e-05
___________ test_batch_processing_results[dmri/fa_in_cst.csv-0-WA()] ___________
Traceback (most recent call last):
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/testing/batch_processing/test_batch_processing.py", line 42, in test_batch_processing_results
    assert (get_csv_float_value(csv_output, row, column) == pytest.approx(  # Default rel_tolerance: 1e-6
AssertionError: assert 0.780605541771859 == 0.7793999429926204 ± 7.8e-07
  +0.780605541771859
  -0.7793999429926204 ± 7.8e-07
___________ test_batch_processing_results[dmri/fa_in_cst.csv-1-WA()] ___________
Traceback (most recent call last):
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/testing/batch_processing/test_batch_processing.py", line 42, in test_batch_processing_results
    assert (get_csv_float_value(csv_output, row, column) == pytest.approx(  # Default rel_tolerance: 1e-6
AssertionError: assert 0.77461183127549 == 0.7755359256476565 ± 7.8e-07
  +0.77461183127549
  -0.7755359256476565 ± 7.8e-07
```

So, we're seeing meaningfully different batch processing results (despite this branch [having passed a few days ago](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/4241779744?check_suite_focus=true)). 

I assume this is an upstream change, and unrelated to this PR, but I'll investigate.

Reply:
My best guess is that this maybe a... one-off glitch? Currently, `test_batch_processing.sh` is [passing on `master`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/4255751692?check_suite_focus=true), too. 

So, the changes from switching `pip` versions shouldn't have affected anything...

Reply:
Still, the fact that a one-off glitch could affect batch processing results is quite curious to me... 

I tried [diffing the `batch_processing.sh` logs](https://www.diffchecker.com/9jqmFWwV), and the only meaningful change I noticed was [in the T2 registation](https://www.diffchecker.com/3Ih4BiJj). But, the T2 CSA results were unaffected! It was the MT and dMRI results that changed. So... I'm at a bit of a loss.

I think the best course of action would be to wait and see if [re-running the processing](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/4267235151?check_suite_focus=true) fixes things, and just... moving on? :thinking:

Reply:
Yep! Restarting the job fixed it. How odd.

Reply:
> But, the T2 CSA results were unaffected! 

indeed, the T2 CSA does not use the PAM50 template registration (it only uses the segmentation and the vertebral levels-- which in this case match that one from the registered PAM50_levels)

one thing concerning is that the straightening produced different results:
![image](https://user-images.githubusercontent.com/2482071/142669565-a6a5cc43-550f-463f-9b6e-0db83336baee.png)

that would explain the PAM50 registration difference (and different MT and DTI results)-- this is strange... i'm pretty sure there is no stochastic computation for the straightening

Reply:
I'm glad that the restarted job shows precisely the same results as before, at least. :thinking:

Reply:
Or... not? It [happened again](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/4267499139?check_suite_focus=true) after merging master once more. This is quite odd.

Reply:
My best guess is that the error is unrelated to the changes in this PR, and we happened to get unlucky with a broken VM that failed in the same way twice in a row, with all other VMs functioning normally.

I'm going to leave any follow-up investigations for https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3597, in case this pops up again in the future.